### ISSUE TITLE
Favor Treats Fever, Heart Valve Disorder, Vascular Disorder, Arthritic Joint Disease

### ISSUE THREAD

**Issue by sstemann** [NCATSTranslator/Feedback#135]
Steps:

1.  CI UI - what drugs may treat Takayasu Arteritis:  https://ui.ci.transltr.io/results?q=c5002e23-c011-49f8-b74f-dc19f4bd1779

Second highest scored result is Favor

![image](https://user-images.githubusercontent.com/38321826/220971794-69594706-39a1-4b30-8e64-10485149390d.png)

2.  Look it up in ARAX https://arax.ncats.io/?r=c5002e23-c011-49f8-b74f-dc19f4bd1779
3. Looks like it's coming from BTE
 
![image](https://user-images.githubusercontent.com/38321826/220972478-d04d18fd-3423-4844-ae48-ccc85760b496.png)

Favor the node doesn't seem to synonymize, i dont know how to find it as a chemical entity to verify this is a chemical entity?  

![image](https://user-images.githubusercontent.com/38321826/220972722-906b9a81-0f9d-40b6-ad13-67e8869c2e36.png)

However the supporting pubmeds dont seem to refer to favor as a chemical entity

4.  When you're in BTE results, note that "Metric (substance)" is the second result, again the pubmeds look like they are using metrics as in unit of measure, not a substance

**Comment by andrewsu** [comment]
TLDR: Yikes, this looks to be a particularly terrible job of [NER](https://en.wikipedia.org/wiki/Named-entity_recognition) by SemMedDB.  Possible solutions are described in these issues:
* https://github.com/biothings/BioThings_Explorer_TRAPI/issues/501
* https://github.com/NCATSTranslator/Feedback/issues/100
* additional info from other teams that consume Semmeddb (e.g., RTX-KG2) on why this issue _didn't_ pop up for them would be welcome

----
Some details on the "Favor" (C0309049) case:
* "Favor" (C0309049) is some sort of (obscure?) vet drug?  I can't find more details/info, but it is a valid UMLS ID: https://uts.nlm.nih.gov/uts/umls/concept/C0309049
* [Our version of semmeddb](https://biothings.transltr.io/semmeddb) has 2809 records with C0309049 as either the subject or object [https://biothings.transltr.io/semmeddb/query?q=subject.umls:C0309049 OR object.umls:C0309049](https://biothings.transltr.io/semmeddb/query?q=subject.umls:C0309049%20OR%20object.umls:C0309049)
* Examining the PMIDs for a subset of these records shows that semmed is recognizing the colloquial usage of "favor":
   * https://pubmed.ncbi.nlm.nih.gov/11089648/: "suggesting that the conformational feature favouring the location of a residue in an alpha-helix is different from the one favouring the beta-sheet"
   * https://pubmed.ncbi.nlm.nih.gov/29395602/: "knowledge gain favoring the Simulation group as compared with the Online-only group"
* Pubtator does not make these same NER errors, suggesting the idea of cross-referencing to pubtator might be valuable (https://github.com/biothings/BioThings_Explorer_TRAPI/issues/501).  We could also look at whether semmeddb itself reports poor scores for NER in these cases, which we could then use as a filter.
   * https://www.ncbi.nlm.nih.gov/research/pubtator/?view=docsum&query=11089648
   * https://www.ncbi.nlm.nih.gov/research/pubtator/?view=docsum&query=29395602

Similar issue with "Metric (substance)" (C0699680) -- poor NER
* [https://biothings.transltr.io/semmeddb/query?q=subject.umls:C0699680 OR object.umls:C0699680](https://biothings.transltr.io/semmeddb/query?q=subject.umls:C0699680%20OR%20object.umls:C0699680)
* https://pubmed.ncbi.nlm.nih.gov/10890402/
* https://pubmed.ncbi.nlm.nih.gov/11473000/

**Comment by sstemann** [comment]
@andrewsu i see metric in ARAGORN's response as well (result #78), favor may be in their results as well but maybe not the top 500.

![image](https://user-images.githubusercontent.com/38321826/221982656-9e544d89-f84f-433b-889b-48f52a5725e8.png)

that's my guess but maybe @uhbrar could confirm if Favor is in their result

**Comment by andrewsu** [comment]
As noted in https://github.com/NCATSTranslator/Feedback/issues/100#issuecomment-1478701011, this issue would be eliminated by adding a filter to semmeddb based on the number of unique PMIDs supporting a relationship.

**Comment by andrewsu** [comment]
... and now thanks to @colleenXu, the new SemMedDB API (where edges must have > 5 unique PMIDs in support) can now be queried via TRAPI.  For example, the query below asks for `DiseaseOrPhenotypicFeature`s that are treated by "Favor" `UMLS:C0309049`.  When posting this query to https://api.bte.ncats.io/v1/smartapi/1d288b3a3caf75d541ffaae3aab386c8/query/ (the existing SemmedDB API with no filtering), BTE returns 89 results, including osteoporosis, coronary artery disorder, and bacterial urinary tract infection.  (All of these are false associations.)  In contrast, when posted to https://api.bte.ncats.io/v1/smartapi/ac9c2ad11c5c442a1a1271223468ced1/query/ (the new filtered API), there are zero results returned.


```
{
    "message": {
        "query_graph": {
            "edges": {
                "e01": {
                    "subject": "n0",
                    "object": "n1",
                    "predicates": ["biolink:treats"]
                }
            },
            "nodes": {
                "n0": {
                    "ids": ["UMLS:C0309049"]
                },
                "n1": {
                    "categories": ["biolink:DiseaseOrPhenotypicFeature"]
                }
            }
        }
    }
}
```

**Comment by sierra-moxon** [comment]
from TAQA, Andrew's update on SEMMED:
this is an error in SEMMED, clearly an error. 
Andrew hypothesized that filtering by number of PMIDs, service provider set up a new service, annotated with smart-api, queried by TRAPI.
See comment above for summary presented; this fixes this ticket! :)

**Comment by sierra-moxon** [comment]
From TAQA w/re to the SEMMEDDB hackathon summary given there:
We can clean up some of the SEMMEDDB results, but not all of them. hesitate to go down the rabbit hole of fixing an unmaintained resource. Two more things can be tried: Andrew has tickets (link to UMLS, look at scores provided by SEMMEDDB). From Bill B and Andrew S: in general, we agree that Text Mined data should be treated differently, with the goal of increasing user trust. Is this a plan for the UI?
From Andy:
if UI gets 3 PMIDs from 3 sources - one is DrugBank, 3 are SEMMEDDB - do we treat those differently? In discussion.
From Sui and Tyler:
papers are evidence, but not the only kind. - numbers of pubs aren't the only factor to provide evidence.
Andrew:
the key evidence here is SEMMEDDB has paper references and so does DrugBank. We don't have the knowledge level to disambiguate the difference between the confidence of these two sources.
If the evidence is only SEMMEDDB -- then maybe a visual queue - like a dotted line, would be helpful.

Please see notes from the TAQA meeting for more discussion here. There is nothing more specific to this issue, but the issue is a use case for the more broad discussion of how to show users SEMMEDDB is being used.

**Comment by andrewsu** [comment]
closing this issue in favor of https://github.com/NCATSTranslator/Feedback/issues/413, which addresses the general issue of NER issues in semmeddb
